File: 0020-0029/0027.rb
require 'date'

wday_lst = %w(Sunday Monday Tuesday Wednesday Thursday Friday Saturday)

while gets
  m,d = $_.split(' ').map(&:to_i)
  break if m == 0 || d == 0
  day = Date::new(2004, m, d)
  puts wday_lst[day.wday]
end

File: python/0027.py
import datetime, sys

wday_lst = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']

for s in sys.stdin:
  m,d = map(int, s.split())
  if m == 0 or d == 0: break
  str = '2004-%d-%d 00:00:00'%(m,d)
  day = datetime.datetime.strptime(str, '%Y-%m-%d %H:%M:%S')
  print wday_lst[day.weekday()]
